File: misc/tarai.pl
sub tarai {
    my ($x, $y, $z) = @_;
    if ($x <= $y) {
	return $y;
    }
    return &tarai(&tarai($x-1,$y,$z),
		  &tarai($y-1,$z,$x),
		  &tarai($z-1,$x,$y));
}
print &tarai(12,6,0),"\n";

File: misc/tarai.rb
def tarai(x,y,z)
  if x <= y then
    return y
  end
  tarai(tarai(x-1,y,z),
        tarai(y-1,z,x),
        tarai(z-1,x,y))
end
print tarai(12,6,0), "\n"
